### ISSUE TITLE
Client-side translation

### ISSUE THREAD

**Issue by rubenwardy** [zeuner/edutest#17]
You currently use intllib, which forces all users to use the same language

I recommend using Minetest's built-in support for translation instead. I wrote a tutorial here: https://rubenwardy.com/minetest_modding_book/en/quality/translations.html

**Comment by zeuner** [comment]
Switching to `minetest.translate` is already on my 2022 agenda because it removes a dependency. However, I'm not sure yet whether the translations will be done client-side, or through `minetest.get_translated_string`. Among the requested functionality I have to consider is the possibility to switch the language while logged in, which seems to be unsupported by Minetest. All accessors for the client language I know of (server-side `minetest.get_player_information` or client-side `minetest.get_language`) are read-only. So I might have to use only parts of Minetest's translation API for now.

**Comment by rubenwardy** [comment]
What's the reason for switching language whilst logged in? Is this an actual request you received? I ask because I'm wondering whether it's because they just couldn't find the language setting. I'm working on a settings GUI redesign which will make it more prominent

Implementing language switching yourself is dubious, only your mod will change language - not other mods or the engine

**Comment by zeuner** [comment]
Yes, the request is to have a language setting right in the EDUtest UI (thus, in-game). To be sure, I pointed out the existing language setting using a screenshot. A downside of it might be that it requires restarting Minetest (https://github.com/minetest/minetest/blob/master/src/settings_translation_file.cpp#L194).

I agree that a mod-specific language switcher should be avoided if possible. Should I have to implement it, I might look into writing a Minetest PR for a language switching API, for which the mod-specific switcher could become a frontend later on.

**Comment by zeuner** [comment]
In the `edutest-2022` branch I have translation support using Minetest's own API. I need to compute most translations server-side (`minetest.get_translated_string`), though, because I need to estimate their rendered size for the layout. Improving this would require a possibility to compute formspec layouts client-side.
